Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -911,5 +918,32 @@ public static class Extensions {
             }
             return prettyPrinted.ToString().Trim();
         }
+
+        public static void RunSynchronouslyOnUIThread(this System.Threading.Tasks.Task task, IServiceProvider serviceProvider, [CallerMemberNameAttribute] string callerMemberName = "") {
+            if (!Dispatcher.CurrentDispatcher.CheckAccess()) {
+                string message = string.Format("{0} must be called on a UI thread.", callerMemberName);
+                Debug.Fail(message);
+                return;
```

Comment:
So is the difference from `WaitAndUnwrapExceptions` that is _does_ allow other tasks to interrupt the one being waited? Or that it _doesn't_?